nat-lab/tests/test_mesh_firewall.py

```diff
@@ -525,3 +525,282 @@ async def on_stdout(stdout: str) -> None:
 
         assert alpha_conn_tracker.get_out_of_limits() is None
         assert beta_conn_tracker.get_out_of_limits() is None
+
+
+@pytest.mark.asyncio
+async def test_mesh_firewall_tcp_stuck_in_last_ack_state_conn_kill_from_server_side() -> None:
+    async with AsyncExitStack() as exit_stack:
+        api = API()
+        (alpha, beta) = api.default_config_two_nodes()
+
+        CLIENT_ALPHA_IP = alpha.ip_addresses[0]
+        CLIENT_BETA_IP = beta.ip_addresses[0]
+        PORT = 12345
+
+        beta.set_peer_firewall_settings(alpha.id, allow_incoming_connections=False)
+
+        (connection_alpha, alpha_conn_tracker) = await exit_stack.enter_async_context(
+            new_connection_with_conn_tracker(
+                ConnectionTag.DOCKER_CONE_CLIENT_1,
+                generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_CONE_CLIENT_1,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+            )
+        )
+        (connection_beta, beta_conn_tracker) = await exit_stack.enter_async_context(
+            new_connection_with_conn_tracker(
+                ConnectionTag.DOCKER_CONE_CLIENT_2,
+                generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_CONE_CLIENT_2,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+            )
+        )
+
+        client_alpha = await exit_stack.enter_async_context(
+            telio.Client(
+                connection_alpha, alpha, telio.AdapterType.BoringTun
```

Comment:
I didn't realize this last time but I think we should expect to see same behaviour on both boring tun and linux native adapters. If that is the case can you parametrize both tests?

Reply:
Do we have proper firewall config in nat-lab tests? Because firewall hooks only properly exist for boringtun.